File: C/Algorithms/654.c
#include <stdlib.h>

//  Definition for a binary tree node.
struct TreeNode {
  int val;
  struct TreeNode *left;
  struct TreeNode *right;
};
typedef struct TreeNode Node;

Node *build_tree(int *nums, int start, int end) {
  int max = *(nums + start), id = start;
  for (int i = start + 1; i < end; i++) {
    int v = *(nums + i);
    if (v > max) {
      max = v;
      id = i;
    }
  }

  // Go left and right:
  Node *left = NULL, *right = NULL;
  if (id > start)
    left = build_tree(nums, start, id);
  if (end > id + 1)
    right = build_tree(nums, id + 1, end);

  Node *n = malloc(sizeof(*n));
  n->val = max;
  n->left = left;
  n->right = right;
  return n;
}

Node *constructMaximumBinaryTree(int *nums, int numsSize) {
  return build_tree(nums, 0, numsSize);
}

File: Javascript/Algorithms/654.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
let constructMaximumBinaryTree = nums => {
    const build_tree = (start, end) => {
        let max = nums[start], id = start;
        for (let i = start+1; i < end; i++){
            let v = nums[i];
            if (v > max){
                max = v;
                id = i;
            }
        }
    
        // Go left and right:
        let left = null, right = null;
        if (id > start)
            left = build_tree(start, id);
        if (end > id + 1)
            right = build_tree(id+1, end);
    
        return new TreeNode(max, left, right);
    };
    return build_tree(0, nums.length)
};
